base/promotion.jl

```diff
@@ -352,6 +352,6 @@ min(x::Real) = x
 max(x::Real) = x
 minmax(x::Real) = (x, x)
 
-max{T<:Real}(x::T, y::T) = ifelse(y < x, x, y)
-min{T<:Real}(x::T, y::T) = ifelse(y < x, y, x)
+max{T<:Real}(x::T, y::T) = select_value(y < x, x, y)
+min{T<:Real}(x::T, y::T) = select_value(y < x, y, x)
```

Comment:
Now that the other methods of `ifelse` are deprecated, I suppose we could replace `ifelse` and `select_value` with a single `ifelse` builtin.